### PR TITLE
IPData API CLI draft

### PR DESCRIPTION
**Pull request by stasdavydov**

### CODE REVIEW

**Review by jonathan-kosgei — CHANGES_REQUESTED** [review]
Let's rename the "--api_key" param to "--api-key"

**Review by jonathan-kosgei — CHANGES_REQUESTED** [review]
Let's show a cleaner message without the traceback when a user calls any of the commands before setting up an API key eg.

```
ipdata info
Please specify IPData API Key
Traceback (most recent call last):
  File "/home/jonathan/.local/bin/ipdata", line 8, in <module>
    sys.exit(cli())
  File "/home/jonathan/.local/lib/python3.6/site-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/home/jonathan/.local/lib/python3.6/site-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/home/jonathan/.local/lib/python3.6/site-packages/click/core.py", line 1137, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/jonathan/.local/lib/python3.6/site-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/jonathan/.local/lib/python3.6/site-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/home/jonathan/.local/lib/python3.6/site-packages/ipdata/cli.py", line 140, in info
    res = IPData(get_and_check_api_key(api_key)).lookup('8.8.8.8')
  File "/home/jonathan/.local/lib/python3.6/site-packages/ipdata/cli.py", line 50, in get_and_check_api_key
    raise WrongAPIKey
ipdata.cli.WrongAPIKey
```

**Comment by jonathan-kosgei** [comment]
Seems to be a bug here

```
ipdata init test
Please specify IPData API Key
New API Key is saved to /home/jonathan/.ipdata
```

It asks to specify the key when I have already done so.

**Comment by jonathan-kosgei** [comment]
There seems to be a bug when getting the users ip address;

```
ipdata myip
Error: IP address 192.168.43.207 is a private IP Address
```
The IP should be retrieved by making a call to `https://api.ipdata.co/ip`

**Comment by jonathan-kosgei** [comment]
For `ipdata ip 1.1.1.1` is it possible to drop the `ip` sub-command so that users can just do `ipdata 1.1.1.1` to get the information for that IP?

**Comment by jonathan-kosgei** [comment]
Instead of pretty-printing the response we should just output the json as is so that users can pipe it to `jq` or run other bash commands on the response.

**Comment by jonathan-kosgei** [comment]
Let's change the `myip` command to `me` and if possible running the `ipdata` command with no parameters should print the user's location information.

**Comment by stasdavydov** [comment]
> There seems to be a bug when getting the users ip address;
> 
> ```
> ipdata myip
> Error: IP address 192.168.43.207 is a private IP Address
> ```
> 
> The IP should be retrieved by making a call to `https://api.ipdata.co/ip`

Should I add related method to IPData class?

**Comment by jonathan-kosgei** [comment]
The functionality is already available;

```
from ipdata import ipdata
from pprint import pprint

ipdata = ipdata.IPData('test')
response = ipdata.lookup(select_field='ip')
pprint(response)
```

**Comment by stasdavydov** [comment]
The `click` library has its restrictions, so I changed the CLI arguments and options the following way to fit what we need.
`cli` -- alias of `cli me`
`cli --ip 8.8.8.8` -- will lookup for 8.8.8.8
`cli --fields country_code` -- will lookup for self IP address and dump country_code only
`cli me` -- will lookup for self IP address
`cli --api-key <key> init` -- will init the API Key.
Every `cli` command can have passed `--api-key <key>` instead of using pre-initialized one.

**Comment by stasdavydov** [comment]
> For `ipdata ip 1.1.1.1` is it possible to drop the `ip` sub-command so that users can just do `ipdata 1.1.1.1` to get the information for that IP?

Click library has limitations. I fixed this with `ipdata --ip 1.1.1.1`. There is no way to have multiple commands with multiple arguments with default (no-command) execution.

**Comment by jonathan-kosgei** [comment]
> > For `ipdata ip 1.1.1.1` is it possible to drop the `ip` sub-command so that users can just do `ipdata 1.1.1.1` to get the information for that IP?
> 
> Click library has limitations. I fixed this with `ipdata --ip 1.1.1.1`. There is no way to have multiple commands with multiple arguments with default (no-command) execution.
> Or I should not use Click to support all of the requirements.

Is it possible to bypass click for this one command?

**Comment by stasdavydov** [comment]
Batch command is implemented.

File: ipdata/cli.py [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+                  f'It will be overwritten with {api_key}',
+                  file=stderr)
+
+        with open(key_path, 'w') as f:
+            f.write(api_key)
+        print(f'New API Key is saved to {key_path}')
+    else:
+        print(f'Failed to check the API Key (Error: {res["status"]}): {res["message"]}',
+              file=stderr)
+
```
Comment (jonathan-kosgei) [inline ipdata/cli.py:92]: I get this error when trying to set a new api key;

```
> ipdata init test
Usage: ipdata init [OPTIONS]
Try "ipdata init --help" for help.

Error: Got unexpected extra argument (test)
```

"test" is a valid API key.

**Comment by stasdavydov** [comment]
Init command is fixed.
Documentation is added.

**Review by jonathan-kosgei — APPROVED** [review]

**Pull request merged**